ext/zpages/test/tracez_data_aggregator_benchmark.cc

```diff
@@ -0,0 +1,249 @@
+ * different latencies.
+ */
+void StartEndSpansLatency(
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  opentelemetry::trace::StartSpanOptions start;
+  start.start_steady_time = SteadyTimestamp(nanoseconds(0));
+  for (; i > 0; i--)
+  {
+    // Latency bucket depends on the index
+    auto latency_band = kLatencyBoundaries[i % kLatencyBoundaries.size()];
+    opentelemetry::trace::EndSpanOptions end;
+    end.end_steady_time = SteadyTimestamp(latency_band);
+
+    tracer->StartSpan(isUnique ? std::to_string(i) : "", start)->End(end);
+  }
+}
+/*
+ * Helper function that creates and ends spans instantly, while simulating
+ * error codes.
+ */
+void StartEndSpansError(
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  for (; i > 0; i--)
+    tracer->StartSpan(isUnique ? std::to_string(i) : "")
+        ->SetStatus(opentelemetry::trace::CanonicalCode::CANCELLED, "");
+}
+
+/*
+ * Helper function that creates i spans that always run.
+ */
+void StartSpans(
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> &spans,
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  for (; i > 0; i--)
+    spans.push_back(tracer->StartSpan(isUnique ? std::to_string(i) : ""));
+}
+
+/*
+ * Helper function that queries for Tracez data. This simulates a user visiting
+ * or refreshing the Tracez webpage many times, except this function does and
+ * holds nothing in memory. While this runs, aggregation work and performance is
+ * expected is be affected negatively.
+ */
+
+void GetManyAggregations(std::unique_ptr<TracezDataAggregator> &aggregator, int i)
+{
+  for (; i > 0; i--)
+    aggregator->GetAggregatedTracezData();
+}
+
+////////////////////////  FIXTURE FOR SHARED SETUP CODE ///////////////////
+
+class TracezAggregator : public benchmark::Fixture
+{
+protected:
+  void SetUp(const ::benchmark::State& state)
+  {
+    std::shared_ptr<TracezSpanProcessor> processor(new TracezSpanProcessor());
+    tracer     = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor));
+    aggregator = std::unique_ptr<TracezDataAggregator>(
+        new TracezDataAggregator(processor, milliseconds(10)));
+  }
+
+  std::unique_ptr<TracezDataAggregator> aggregator;
+  std::shared_ptr<opentelemetry::trace::Tracer> tracer;
+};
+
+//////////////////////////// BENCHMARK DEFINITIONS /////////////////////////////////
+
+//////////////////////// USER NEVER VISITS WEBPAGE //////////////////////////////
+
+/*
+ * Aggregator handing many spans with the same name, who end instantly. This
+ * checks the scenario where there's only one Tracez name and minimal sorting
+ * of latencies is required, as all spans should be sorted in the same bucket
+ * under the same span name.
+ */
+BENCHMARK_DEFINE_F(TracezAggregator, BM_InstantSame)(benchmark::State &state)
```

Comment:
Nit: Consider updating the names of the benchmarks to better reflect what the benchmark is doing.
For example here the name can be changed to: AggregateShortLatencySpans

Reply:
that makes sense, i'll update all the names!!